lib/cast.js

```diff
@@ -63,9 +63,7 @@ exports.schema = function (config) {
         return internals.any.valid(config);
     }
 
-    Hoek.assert(config === null, 'Invalid schema content:', config);
-
-    return internals.any.valid(null);
```

Comment:
For the exact same reason as what i answered on your first comment.

In this PR ``null`` and ``undefined`` are volontary not the same, because there are existing unit test that already treat it differently on the ``null`` case.

But i have to say that i don't really get this choice...